lib/Dialect/FIRRTL/Import/FIRParser.cpp

```diff
@@ -1491,33 +1491,42 @@ struct LazyLocationListener : public OpBuilder::Listener {
 
 /// Remove the DontTouch annotation and return true, if it exists.
 /// Ignore DontTouch that will apply only to a subfield.
-static bool removeDontTouch(ArrayAttr &annotations) {
+/// Return true if a "circt.nonlocal" annotation found and the checkNonLocal is
+/// set.
+static bool removeDontTouch(ArrayAttr &annotations, bool checkNonLocal = true) {
   SmallVector<Attribute> filteredAnnos;
-  bool hasDontTouch = false;
+  bool hasDontTouch = false, hasNonLocal = false;
```

Comment:
I think this can be changed to `bool needsSymbol = false;` instead of tracking separately dont touch vs. non-local and then returning the `OR` of the two bools.

Reply:
Done